torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1166,6 +1169,9 @@ def summon_full_params(
                 discarded after the context manager exists;
                 disabling this can be slightly more efficient (default: True)
         """
+        # Monkey patch `named_parameters()`
+        torch_named_parameters = torch.nn.Module.named_parameters
+        self.named_parameters = self._fsdp_named_parameters  # type: ignore[assignment]
```

Comment:
Adding the `type: ignore[assignment]` seems to be the best solution. See https://github.com/python/mypy/issues/2427.